src/components/common/Dialogs/CreateDomainDialog/CreateDomainDialogForm.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { ColonyRole, Id } from '@colony/colony-js';
+import { defineMessages } from 'react-intl';
+import { useFormContext } from 'react-hook-form';
+
+import ColorSelect from '~shared/ColorSelect';
+import {
+  ActionDialogProps,
+  DialogControls,
+  DialogHeading,
+  DialogSection,
+} from '~shared/Dialog';
+import { HookFormInput as Input, Annotations } from '~shared/Fields';
+import NoPermissionMessage from '~shared/NoPermissionMessage';
+import PermissionRequiredInfo from '~shared/PermissionRequiredInfo';
+// import NotEnoughReputation from '~dashboard/NotEnoughReputation';
+
+import { useDialogActionPermissions } from '~hooks'; // useEnabledExtensions
+
+import styles from './CreateDomainDialogForm.css';
+
+const displayName = 'common.CreateDomainDialog.CreateDomainDialogForm';
+
+const MSG = defineMessages({
+  titleCreate: {
+    id: `${displayName}.titleCreate`,
+    defaultMessage: 'Create a new team',
+  },
+  name: {
+    id: `${displayName}.name`,
+    defaultMessage: 'Team name',
+  },
+  purpose: {
+    id: `${displayName}.name`,
+    defaultMessage: 'What is the purpose of this team?',
+  },
+  annotation: {
+    id: `${displayName}.annotation`,
+    defaultMessage: 'Explain why you’re creating this team (optional)',
+  },
+  cannotCreateMotion: {
+    id: `${displayName}.cannotCreateMotion`,
+    defaultMessage: `Cannot create motions using the Governance v{version} Extension. Please upgrade to a newer version (when available)`,
+  },
+});
+
+const CreateDomainDialogForm = ({ back, colony }: ActionDialogProps) => {
+  const {
+    formState: { isValid, isSubmitting },
+  } = useFormContext();
+
+  // const {
+  //   votingExtensionVersion,
+  //   isVotingExtensionEnabled,
+  // } = useEnabledExtensions({
+  //   colonyAddress: colony.colonyAddress,
+  // });
+
+  const requiredRoles: ColonyRole[] = [ColonyRole.Architecture];
+  const [userHasPermission, onlyForceAction] = useDialogActionPermissions(
+    colony,
+    false, // isVotingExtensionEnabled,
+    requiredRoles,
+    [Id.RootDomain],
+  );
+
+  const inputDisabled = !userHasPermission || onlyForceAction || isSubmitting;
+
+  // const cannotCreateMotion =
+  //   votingExtensionVersion ===
+  //     VotingReputationExtensionVersion.FuchsiaLightweightSpaceship &&
+  //   !values.forceAction;
+
+  return (
+    <>
+      <DialogSection appearance={{ theme: 'sidePadding' }}>
+        <DialogHeading title={MSG.titleCreate} />
+      </DialogSection>
+      {!userHasPermission && (
+        <DialogSection>
+          <PermissionRequiredInfo requiredRoles={[ColonyRole.Architecture]} />
+        </DialogSection>
+      )}
+      <DialogSection>
+        <div className={styles.nameAndColorContainer}>
+          <div className={styles.domainName}>
```

Comment:
I see that this combination of domain and color select is also included in the EditDomainDialogForm, so I think could be moved to own component.